crates/interfaces/src/executor.rs

```diff
@@ -1,50 +1,55 @@
+use crate::RethError;
 use reth_primitives::{BlockNumHash, Bloom, PruneSegmentError, B256};
+use revm_primitives::EVMError;
 use thiserror::Error;
 
 /// Transaction validation errors
-#[allow(missing_docs)]
 #[derive(Error, Debug, Clone, PartialEq, Eq)]
 pub enum BlockValidationError {
     /// EVM error with transaction hash and message
-    #[error("EVM reported invalid transaction ({hash:?}): {message}")]
+    #[error("EVM reported invalid transaction ({hash}): {error}")]
     EVM {
         /// The hash of the transaction
         hash: B256,
-        /// Error message
-        message: String,
+        /// The EVM error.
+        #[source]
+        error: Box<EVMError<RethError>>,
     },
     /// Error when recovering the sender for a transaction
-    #[error("Failed to recover sender for transaction")]
+    #[error("failed to recover sender for transaction")]
     SenderRecoveryError,
     /// Error when incrementing balance in post execution
-    #[error("Incrementing balance in post execution failed")]
+    #[error("incrementing balance in post execution failed")]
     IncrementBalanceFailed,
     /// Error when receipt root doesn't match expected value
-    #[error("Receipt root {got:?} is different than expected {expected:?}.")]
+    #[error("receipt root {got} is different than expected {expected}")]
     ReceiptRootDiff {
         /// The actual receipt root
-        got: B256,
+        got: Box<B256>,
         /// The expected receipt root
-        expected: B256,
+        expected: Box<B256>,
     },
```

Comment:
why did these fields get boxed?

Reply:
reduces size, I'll open another PR with more of this